designsafe/apps/api/projects/views.py

```diff
@@ -595,25 +533,35 @@ def get(self, request, project_id=None, name=None, uuid=None):
     @profile_fn
     def delete(self, request, uuid):
         """
+        Delete metadata. This will delete associated metadata objects.
+        This should not be used to delete a project (only metadata related
+        to a project).
 
         :return:
         :rtype: JsonResponse
         """
-        ag = get_service_account_client()
-        meta_obj = ag.meta.getMetadata(uuid=uuid)
-        model = project_lookup_model(meta_obj)
-        meta = model(**meta_obj)
-        ag.meta.deleteMetadata(uuid=uuid)
+        try:
+            client = request.user.agave_oauth.client
```

Comment:
@Strmiska , it seems like the delete here fails correctly if a user doesn't have access to the project 👍  

so you are saying that the try/except block in `put` would disallow a user from changing things if they weren't supposed to. Then, 👍 .  and i assume for the `post`, there is a similar safeguard (but one that i can't recognize as i'm not familiar here...but thinking it is the listing that would fail).

Reply:
@nathanfranklin Thanks for double checking some of this! I left myself a few notes on this PR that I'm going to double check over today. It's been a while since anyone really dug through some of these methods, but the only reason we use the `service_account` on these endpoints should be to create, delete, and update user permissions on Tapis meta objects. I'm going to comb over this one more time to double check that nothing is exclusively running under the `service_account`.